### ISSUE TITLE
No valid To: header in reply

### ISSUE THREAD

**Issue by rgrcnh** [SIPp/sipp#152]
I made a small patch to avoid an early exit of sipp, when the sipp receives a not well formed SIP message. On that situation, SIPP shows  "No valid To: header in reply" on console and quits.  With the patch the message is discarded even before the `To:` test on sipMsgCheck method.
I hope someone enjoy this.

    $ diff -crB  ../../sipp-vanilla/sipp-3.4.1/src/socket.cpp socket.cpp 
    *** ../../sipp-vanilla/sipp-3.4.1/src/socket.cpp  2014-03-09 16:04:57.000000000 -0300
    --- socket.cpp  2015-06-22 16:09:16.853990854 -0300
    ***************
    *** 1079,1084 ****
    --- 1079,1086 ----
  
      void process_message(struct sipp_socket *socket, char *msg, ssize_t msg_size, struct sockaddr_storage *src)
      {
    +     char *hdr = NULL;
    + 
          // TRACE_MSG(" msg_size %d and pollset_index is %d \n", msg_size, pollset_index));
          if(msg_size <= 0) {
              return;
    ***************
    *** 1091,1096 ****
    --- 1093,1104 ----
              return;
          }
  
    +     if(!hdr) hdr = strcasestr((char *) msg, "\r\nto:");
    +     if(!hdr) hdr = strcasestr((char *) msg, "\r\nt:");
    +     if(!hdr) { 
    +       WARNING("SIP nom standart");
    +        return;
    +     }
          char *call_id = get_call_id(msg);
          if (call_id[0] == '\0') {
              WARNING("SIP message without Call-ID discarded");

**Comment by vodik** [comment]
What is the exact header you're getting which SIPp is claiming to be invalid? Because looking at the implementation of `get_header`, it looks like we already are doing case-insensitive comparisons and also attempting to handle the short hand headers (e.g. "t:"). I'd like to see if I can reproduce the situation first to understand it.

I appreciate the effort, but that warning isn't really correct, considering that the "t:" style header is completely valid and defined in rfc3261. If we're not handling them correctly, that's a bug!

**Comment by rgrcnh** [comment]
Vodik,
The header arrives without the field "To:", from the PBX I am testing.
I my case, when I was testing the device (PBX) at its limit, I started to receive packets not well formatted, after injecting 250 calls per second, and 15000 calls on the system, on some answers. This behavior (from the pbx) is causing sipp stop the test instead compute a received error.
Looking at SIPP code, probably it is assuming that all arriving packet (with "SIP/2.0" at the header) will be correct and well formatted on all fields. The simple code I wrote is taking care of a packet that arrived without `To:` field, but sipp needs (probably) a routine to check if all required fields are present before any parse, making itself safe from any bad SIP implementations.
[ ]'s
